tests/js/client/aql/aql-index-batch-materialize.js

```diff
@@ -465,6 +465,31 @@ function IndexBatchMaterializeTestSuite() {
       }
     },
 
+    testMaterializationNonUniqueIndexWithOption: function() {
+      // should behave the same as testMaterializationUniqueIndex even though the used attribute is not a unique index
+      // the false case is covered by testMaterializationNonUniqueIndex
+      const query = `
+        FOR dx IN ${singleShardCollection}
+          FOR dy IN ${singleShardCollection} OPTIONS { pushDownMaterialization: true }
+            FILTER dx.x > 1 AND dx.x == dy.x
+            RETURN [dx, dy]
+      `;
+      const { firstIndexNode, secondIndexNode, firstMaterializeNode, secondMaterializeNode }
+          = parseExplainationAndExecuteQuery(query);
+
+      if (internal.isEnterprise() || !internal.isCluster()) {
+        // it should be index->index->materialize->materialize
+        assertEqual(firstIndexNode.id, secondIndexNode.dependencies[0]);
+        assertEqual(secondIndexNode.id, firstMaterializeNode.dependencies[0]);
+        assertEqual(firstMaterializeNode.id, secondMaterializeNode.dependencies[0]);
+      } else {
+        // Cluster on community version behaves differently and does not apply this improvment.
+        // should behave the same as `testMaterializationTupleUniqueIndex`
+        assertEqual(firstIndexNode.id, firstMaterializeNode.dependencies[0]);
+        assertEqual(secondIndexNode.id, secondMaterializeNode.dependencies[0]);
+      }
+    },
+
```

Comment:
To make sure that this remains a valid test, I want the same query with `pushDownMaterialization=false` and assert that the loops are not moved past each other.